### ISSUE TITLE
[BUG] checkpoint.load catches bare Exception and rewraps as Model — masks root cause

### ISSUE THREAD

**Issue by sachncs** [sachncs/morel#28]
## Problem

`morel.train.checkpoint.load(target)` catches **any** `Exception` from `torch.load(..., weights_only=True)` and re-raises as `Model(f"checkpoint at {path} could not be loaded safely: {exc}")` at `morel/train/checkpoint.py:39-42`. The error message lumps together:

- genuine `Model` concerns (file not a dict, unknown keys)
- I/O concerns (`FileNotFoundError` is re-raised separately on line 38 — fine — but `IsADirectoryError`, `PermissionError`, `OSError` get rewrapped)
- pytorch contract changes (`RuntimeError` from a future pytorch version that tightens `weights_only=True`)
- every other `Exception`

The rewrap loses the original traceback for everything except `Model`, and the new error type hides the root cause from anything that matches on `Model`.

## Evidence

- `morel/train/checkpoint.py:39-42`

```python
try:
    payload = torch.load(path, map_location="cpu", weights_only=True)
except Exception as exc:
    raise Model(f"checkpoint at {path} could not be loaded safely: {exc}") from exc
```

- `tests/unit/train/checkpoint.py` — currently asserts `Model` is raised on unknown keys, which is fine; no test exercises the broad-except path.
- `morel/train/checkpoint.py:51-63` (`unsafe`) does the same broad-except by *not* catching, which is also a divergence — two load functions with different error semantics on identical inputs.

## Root Cause

The check uses `try/except Exception` instead of the more selective `RuntimeError` or `(UnpicklingError, RuntimeError, OSError)`.

## Impact

- Debugging a corrupted checkpoint requires unwrapping a `Model` whose `.args[0]` is a stringified exception, with the original traceback preserved only via `__cause__`.
- A future pytorch change that raises a previously-unseen exception class would be silently swallowed as `Model` instead of bubbling up.
- Inconsistent error behaviour between `load` and `unsafe` complicates library consumers.

## Recommendation

- Narrow the `except` to `(UnpicklingError, RuntimeError, OSError)` and let other exceptions bubble up. Keep the `Model` rewrap only for the dict / unknown-keys checks.
- Add `__cause__` explicitly (`raise Model(...) from exc`) — already done — but also surface the original class name in the message so operators can grep for it.

## Acceptance Criteria

- [ ] A corrupt `.pt` file produces an error whose `__cause__` is the original torch exception and whose message names that exception class.
- [ ] An I/O error (`PermissionError`) is re-raised as the original exception class, not `Model`.
- [ ] `load` and `unsafe` document and test their divergent error semantics.

## Verification

- Add `tests/unit/train/checkpoint.py::Checker::load_wraps_runtime_error` that writes a malformed `.pt` and asserts the cause is preserved.
- `pytest tests/ -q` remains green.

## Dependencies

None.

## Risk

Narrowing the `except` may surface currently-swallowed exceptions in unrelated callers. Those callers should either fix the root cause or catch explicitly.

**Comment by sachncs** [comment]
Resolved in master. The relevant fix is split across one or two commits; see the change history since the issue was opened.
